File: 1567-maximum-number-of-vowels-in-a-substring-of-given-length/maximum-number-of-vowels-in-a-substring-of-given-length.cpp
class Solution {
public:
    int maxVowels(string s, int k) {
        int maxi;
        int count=0;
        for(int l=0;l<k;l++){
            if(s[l]=='a'||s[l]=='e'||s[l]=='i'||s[l]=='o'||s[l]=='u'){
                count++;
            }
        }
        maxi=count;
        
        for(int h=k;h<s.length();h++){
            if(s[h-k]=='a'||s[h-k]=='e'||s[h-k]=='i'||s[h-k]=='o'||s[h-k]=='u'){
                count--;
            }
             if(s[h]=='a'||s[h]=='e'||s[h]=='i'||s[h]=='o'||s[h]=='u'){
                count++;
            }
             maxi=max(maxi,count);
        }
        return maxi;
    }
};

File: 1567-maximum-number-of-vowels-in-a-substring-of-given-length/maximum-number-of-vowels-in-a-substring-of-given-length.java
class Solution {
    public boolean vowel(char ch){
        return ch=='a'||ch=='e'||ch=='i'||ch=='o'||ch=='u';
    }
    public int maxVowels(String s, int k) {
        int count=0;
        int max=0;
        for(int i=0;i<k;i++){
            char ch=s.charAt(i);
            if(vowel(s.charAt(i))){
                count++;
            }
        }
        max=count;
        for(int i=k;i<s.length();i++){
            if(vowel(s.charAt(i))){
                count++;
            }
            if(vowel(s.charAt(i-k))){
                count--;
            }
            max=Math.max(max,count);
        }
        return max;
    }
}